GUIDE.md

```diff
@@ -0,0 +1,104 @@
+# Parsimmon Guide
+
+## Parsing indentation sensitive languages
+
+Parsing languages with context like Python or Markdown which use indentation to keep track of nesting structure works best with creating a constructor function around your call to `Parsimmon.createLanguage`:
+
+```js
+const createMyLanguage = ({ indentSize }) =>
+  Parsimmon.createLanguage({
+    Indent: () => Parsimmon.string(" ").times(indentSize),
+    ForLoop: l =>
+      l.SomeBlockStart.chain(block => {
+        const lang = createMyLanguage({
+          indentSize: block.newIndentSize
+        });
+        // `lang` here is a different language instance than `l`, which has a
+        // different `indentSize` so it looks for a different level of
+        // indentation. You only need to know about your current indentation
+        // level for everything to work out in this example, but you could
+        // include more information
+        return lang.Item.atLeast(1);
+      })
+  });
+
+const MyLanguage = createMyLanguage({ indentSize: 0 });
+const ast = MyLanguage.File.tryParse(/* ... */);
+console.log(ast);
+```
+
+The general idea is to take context in at the top level and construct new languages inside of your language each time you nest indentation. For a complete example, [please read](examples/python-ish.js).
+
+## TODO: Performance tips
+
+> Some performance tips. Which constructions should be considered dangerous performance-wise?
+
+I want to answer this one, but honestly don't even really know where to start. It's not something I look at much.
```

Comment:
I think it might be worth mentioning that for parsing efficiency it is good to have the preceding parser to eat all the trailing whitespace, so the next one doesn't have to munch through it again.

```js
const token = Parsimmon.regex(/[a-Z]/)

const whitespace1 = Parsimmon.regex(/[ \t]+/)

const parser = Parsimmon.seq(
    token.skip(whitespace1),
    Parsimmon.alt(
        // other parsers
    )
)
```